### PR TITLE
Filters module

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Refactoring

### Description
Introduce a new `kroxylicious-additional-filters` module to maven hierarchy:

1. Externalising the filters which are not required for Kroxylicious to operate but offer compelling functionality.
2. Provides a useful extension point to ensure the additional filters do not depend on core Kroxylicious modules

### Additional Context

I was reviving my Otel filter and realised I was introducing the third filter module and that seemed awkward to just lump into the root pom

### CODE REVIEW

File: kroxylicious-additional-filters/kroxylicious-schema-validation/pom.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
         <groupId>io.kroxylicious</groupId>
         <artifactId>kroxylicious-parent</artifactId>
         <version>0.2.0-SNAPSHOT</version>
-        <relativePath>../pom.xml</relativePath>
+        <relativePath>../../pom.xml</relativePath>
```
Comment (robobario) [inline kroxylicious-additional-filters/kroxylicious-schema-validation/pom.xml:18]: Should this point at the new additional-filters module as parent now?

**Review by robobario — APPROVED** [review]
Good idea, thanks @SamBarker . The other benefit is to people trying to grok the code, having hints in the dir structure like this is helpful.

File: kroxylicious-additional-filters/kroxylicious-schema-validation/pom.xml (hunk 1 above)
Comment (SamBarker) [reply kroxylicious-additional-filters/kroxylicious-schema-validation/pom.xml]: Good catch. That was fine before I added the enforcer rules.

**Comment by SamBarker** [comment]
```
[INFO] Reactor Summary for Kroxylicious Parent 0.2.0-SNAPSHOT:
[INFO] 
[INFO] Kroxylicious Parent ................................ SUCCESS [  2.541 s]
[INFO] Kroxylicious API ................................... SUCCESS [  4.101 s]
[INFO] Code Generation plugin ............................. SUCCESS [  6.437 s]
[INFO] Kroxylicious Filter API ............................ SUCCESS [  5.885 s]
[INFO] Filter Unit Test tooling ........................... SUCCESS [  2.814 s]
[INFO] Kroxylicious Proxy ................................. SUCCESS [  8.613 s]
[INFO] Testing tools ...................................... SUCCESS [  6.818 s]
[INFO] Additional Filters ................................. SUCCESS [  0.083 s]
[INFO] Kroxylicious Multitenant ........................... FAILURE [  0.067 s]
[INFO] Kroxylicious Schema Validation ..................... SKIPPED
[INFO] Kroxylicious Integration tests ..................... SKIPPED
[INFO] Micro benchmarks ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  37.515 s
[INFO] Finished at: 2023-06-07T11:03:56+12:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.3.0:enforce (filters-use-public-api) on project kroxylicious-multitenant: 
[ERROR] Rule 0: org.apache.maven.enforcer.rules.dependency.BannedDependencies failed with message:
[ERROR] io.kroxylicious:kroxylicious-multitenant:jar:0.2.0-SNAPSHOT
[ERROR]    io.kroxylicious:kroxylicious:jar:0.2.0-SNAPSHOT <--- banned via the exclude/include list
```

A deliberately introduced failure to demonstrate.

**Pull request merged**